src/components/AnnotationCreator.jsx

```diff
@@ -19,27 +19,96 @@ import usePostAnnotation from '../models/annotation/usePostAnnotation';
 import Button from './Button';
 import Text from './Text';
 import StandardDialog from './StandardDialog';
-import CustomAlert from './Alert';
+import Alert from './Alert';
+import useAddEncounter from '../models/encounter/useAddEncounter';
+import useAddAnnotationsToSightingEncounter from '../models/encounter/useAddAnnotationsToSightingEncounter';
 
 function percentageToPixels(percentValue, scalar) {
   const pixelValue = 0.01 * scalar * percentValue;
   return Math.round(Math.max(pixelValue, 0));
 }
 
+async function createEncounter(sightingData, addEncounterToSighting) {
+  const copiedProperties = pick(sightingData, [
+    'time',
+    'timeSpecificity',
+    'locationId',
+  ]);
+  const sightingId = get(sightingData, 'guid');
+  const encounterGuidsBeforeCreation = map(
+    get(sightingData, 'encounters'),
+    encounter => encounter.guid,
+    [],
+  );
+  const encounterCreationResponse = await addEncounterToSighting(
+    sightingId,
+    copiedProperties,
+  );
+  const encounterGuidsAfterCreation = map(
+    get(
+      encounterCreationResponse,
+      ['response', 'data', 'encounters'],
+      [],
+    ),
+    encounter => encounter.guid,
+    [],
+  );
+  const newEncounterGuids = filter(
+    encounterGuidsAfterCreation,
+    enc => encounterGuidsBeforeCreation.indexOf(enc) === -1,
+    [],
+  );
+  return encounterCreationResponse?.success &&
+    newEncounterGuids?.length === 1
+    ? newEncounterGuids
+    : [];
+}
+
 export default function AnnotationCreator({
   titleId = 'CREATE_ANNOTATION',
   asset,
   onClose,
   sightingData,
   refreshSightingData,
+  pending,
 }) {
   const [viewpoint, setViewpoint] = useState('');
   const [IAClass, setIAClass] = useState('');
   const [rect, setRect] = useState({});
   const [anchorEl, setAnchorEl] = useState(null);
+  const [errors, setErrors] = useState([]);
```

Comment:
Because you are only using this to store other persisted values, it shouldn't be necessary.
`errors` can just be calculated from data that you already have.